src/main/java/org/sunbird/profile/service/ProfileServiceImpl.java

```diff
@@ -338,6 +351,108 @@ public SBApiResponse profileUpdate(Map<String, Object> request, String userToken
 		return response;
 	}
 
+	private String validateCadreDetails(Map<String, Object> profileDetailsMap) {
+		List<Map<String, Object>> cadreDetails;
+
+		try {
+			if (profileDetailsMap.containsKey(Constants.CADRE_DETAILS)
+					&& profileDetailsMap.get(Constants.CADRE_DETAILS) != null) {
+				JsonNode cadreJson = mapper.valueToTree(profileDetailsMap.get(Constants.CADRE_DETAILS));
+				if (cadreJson.isNull()) {
+					return "CadreDetails is missing";
+				}
+				if (!cadreJson.has(Constants.CIVIL_SERVICE_TYPE) || cadreJson.get(
+						Constants.CIVIL_SERVICE_TYPE).isNull() || StringUtils.isEmpty(
+						cadreJson.get(Constants.CIVIL_SERVICE_TYPE).asText())) {
+					return "Civil service type ID of CadreDetails is missing";
+				}
+				if (!cadreJson.has(Constants.CIVIL_SERVICE_TYPE_ID) || cadreJson.get(
+						Constants.CIVIL_SERVICE_TYPE_ID).isNull() || StringUtils.isEmpty(
+						cadreJson.get(Constants.CIVIL_SERVICE_TYPE_ID).asText())) {
+					return "Civil service type of CadreDetails is missing";
+				}
+				if (!cadreJson.has(Constants.CIVIL_SERVICE_NAME) || cadreJson.get(
+						Constants.CIVIL_SERVICE_NAME).isNull() || StringUtils.isEmpty(
+						cadreJson.get(Constants.CIVIL_SERVICE_NAME).asText())) {
+					return "Service Name of CadreDetails is missing";
+				}
+				if (!cadreJson.has(Constants.CIVIL_SERVICE_ID) || cadreJson.get(Constants.CIVIL_SERVICE_ID)
+						.isNull() || StringUtils.isEmpty(cadreJson.get(Constants.CIVIL_SERVICE_ID).asText())) {
+					return "Civil service ID  of CadreDetails is missing";
+				}
+				if (!cadreJson.has(Constants.CADRE_NAME) || cadreJson.get(Constants.CADRE_NAME).isNull()
+						|| StringUtils.isEmpty(cadreJson.get(Constants.CADRE_NAME).asText())) {
+					return "cadrename of CadreDetails is missing";
+				}
+				if (!cadreJson.has(Constants.CONTROLLING_AUTHORITY) || cadreJson.get(
+						Constants.CONTROLLING_AUTHORITY).isNull() || StringUtils.isEmpty(
+						cadreJson.get(Constants.CONTROLLING_AUTHORITY).asText())) {
+					return "Controlling authority of CadreDetails is missing";
+				}
+				if (!cadreJson.has(Constants.CADRE_BATCH) || cadreJson.get(Constants.CADRE_BATCH).isNull()
+						|| !cadreJson.get(Constants.CADRE_BATCH).isInt()) {
+					return "Cadre Batch of CadreDetails is missing and it should be an integer";
+				}
+				Map<String, Object> propertyMap = new HashMap<>();
+				propertyMap.put(Constants.ID, Constants.CADRE_CONFIG);
+				cadreDetails = cassandraOperation.getRecordsByPropertiesWithoutFiltering(
+						Constants.KEYSPACE_SUNBIRD, Constants.TABLE_SYSTEM_SETTINGS, propertyMap,
+						null, 2);
+				if (cadreDetails.isEmpty() && !cadreDetails.contains(Constants.VALUE)) {
+					return "Details not found for this key";
```

Comment:
error should be -> Failed to validate cadreDetails schema.

Reply:
done